src/qt/pivx/masternodeswidget.cpp

```diff
@@ -197,29 +198,77 @@ void MasterNodesWidget::onEditMNClicked(){
     }
 }
 
-void MasterNodesWidget::startAlias(QString strAlias){
+void MasterNodesWidget::startAlias(QString strAlias) {
     QString strStatusHtml;
     strStatusHtml += "Alias: " + strAlias + " ";
 
     for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
         if (mne.getAlias() == strAlias.toStdString()) {
             std::string strError;
-            CMasternodeBroadcast mnb;
-            if (CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb)) {
-                strStatusHtml += "successfully started.";
-                mnodeman.UpdateMasternodeList(mnb);
-                mnb.Relay();
-                mnModel->updateMNList();
-            } else {
-                strStatusHtml += "failed to start.\nError: " + QString::fromStdString(strError);
-            }
+            strStatusHtml += (!startMN(mne, strError)) ? ("failed to start.\nError: " + QString::fromStdString(strError)) : "successfully started.";
             break;
         }
     }
-    inform(strStatusHtml);
+    // update UI and notify
+    updateModelAndInform(strStatusHtml);
+}
+
+void MasterNodesWidget::updateModelAndInform(QString informText) {
+    mnModel->updateMNList();
+    inform(informText);
+}
+
+bool MasterNodesWidget::startMN(CMasternodeConfig::CMasternodeEntry mne, std::string& strError) {
+    CMasternodeBroadcast mnb;
+    if (!CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb))
+        return false;
+
+    mnodeman.UpdateMasternodeList(mnb);
+    mnb.Relay();
+    return true;
+}
+
+void MasterNodesWidget::onStartAllClicked() {
+    if(!verifyWalletUnlocked()) return;
+    if (isLoading) {
+        inform(tr("Start all Masternodes is being executed, please wait"));
+    } else {
+        isLoading = true;
+        if (!execute(REQUEST_START_ALL)) {
+            inform(tr("Cannot perform start all Mastenodes"));
```

Comment:
shouldn't we reset `isLoading` to `false` here?

Reply:
it gets set to `false` at the end of the `run()` function (line 261 below)

Reply:
Yeah but we are checking failures of execute.
So I was thinking this might be failing even before entering `run`.
But checking `PWidget::execute` it seems that it can only return true (not sure why its return value is bool then).

Reply:
yes, need to set it to false.

`PWidget::execute` method signature is fine, the goal is not have to change every place that implements it every time that the internal implementation is changed (if change),  it's part of the new GUI screen building framework.